.github/workflows/upload.yml

```diff
@@ -36,21 +42,34 @@ jobs:
           REDIS_HOST: 'localhost'
           REDIS_PORT: 6379
 
-      - name: Build
-        run: |
-          docker build -t reg.desk-lab.de/doku -t reg.desk-lab.de/doku:${{ github.sha }} .
-          cd doku/static && docker build -t reg.desk-lab.de/doku-static -t reg.desk-lab.de/doku-static:${{ github.sha }} .
-          docker login https://reg.desk-lab.de --username ${{ secrets.DOCKER_REGISTRY_USER }} --password ${{ secrets.DOCKER_REGISTRY_PASSWORD}}
-          docker push reg.desk-lab.de/doku-static | cat
-          docker push reg.desk-lab.de/doku-static:${{ github.sha }} | cat
-          docker push reg.desk-lab.de/doku | cat
-          docker push reg.desk-lab.de/doku:${{ github.sha }} | cat
+      - name: Login to Registry
+        uses: docker/login-action@v1 
+        with:
+          registry: reg.desk-lab.de
+          username: ${{ secrets.DOCKER_REGISTRY_USER }}
+          password: ${{ secrets.DOCKER_REGISTRY_PASSWORD}}
+
+      - name: Build and push doku-static
+        uses: docker/build-push-action@v2
+        with:
+          context: doku/static
+          push: true
+          tags: |
+            reg.desk-lab.de/doku-static:${{ github.sha }}
+            reg.desk-lab.de/doku-static
+
+      - name: Build and push doku
+        uses: docker/build-push-action@v2
+        with:
+          context: docker
+          push: true
+          tags: |
+            reg.desk-lab.de/doku${{ github.sha }}
+            reg.desk-lab.de/doku
 
-      - name: Deploy
+      - name: Deploy # TODO: Use docker-action?
```

Comment:
With the action `docker/login-action@v1` you have already logged in to the registry, and with the action `docker/build-push-action@v2`, you have already built and **pushed** the container to our registry. Thus, this last step is not necessary.